news-catalyst/google-app-scripts
# Hasura-backed translations for sidebar

Closes #205 and #208

This was complicated and entailed a fair amount of code changes. I have tested a few times with 2 documents, deleting the data and starting over a couple of times, so hopefully, aside from obvious UX issues, this is functional enough for a first pass at testing. Let me know if you run into problems.

To test:

* open the script editor, run > test as add-on > Franny Lou's doc (latest code)
* open the admin tools*
* scroll down, you should see "Click to translate: Spanish" as the oaklyn site supports english & spanish
* click the "Spanish" link, the top of the sidebar should update with a link to edit in spanish
* click the link - this doc should hopefully have the latest version of the sidebar code as I published it just now
* this new doc should have "es -" prepended to the title
* open the admin tools, you should see existing translations include english
* open the publishing tools, you should be able to fill in the spanish translations and preview + publish
* if you have the front-end running locally (branch: `bugfix/preview-locale`), both preview and publish urls should load the content
  * caveat: front-end preview functionality in front-end had a bug which I fixed in [PR#339](https://github.com/news-catalyst/next-tinynewsdemo/pull/339) - run the front-end out of `bugfix/preview-locale` branch

* I wasn't sure where to put the translation links; admin tools or publishing tools. either way they almost def need to go outside of "debugging info" :) (tbd)
